### PR TITLE
Sahra

### PR DESCRIPTION
**Pull request by Sahra-Zhou**

### CODE REVIEW

File: backend/sample.js [hunk 1]
```diff
@@ -76,6 +91,9 @@ app.get("/", async (req, res) => {
   res.send(allFolders);
 });
 // folder page: get all notes
+//I use post methods here because it needs to get password for private folders 
+//and I can't think of an alternative. 
+//Please comment if you have any suggestions or alternative solutions!!!
```
Comment (matthewzelten) [inline backend/sample.js:96]: Ive been looking into a solution for the password for private folders and seem to be running into a similar issue

File: backend/sample.js [hunk 2]
```diff
@@ -84,11 +102,11 @@ app.post("/:folderName", (req, res) => {
     res.status(404).send("Folder not found.");
   } else {
     if (!result["isPrivate"]) {
-      result = result.notes;
+      result = findNotesByFolder(folderName);
       res.status(201).send(result);
     } else {
       if (passw["password"] === result["password"]) {
-        res.status(201).send(result.notes);
+        res.status(201).send(findNotesByFolder(folderName));
```
Comment (matthewzelten) [inline backend/sample.js:109]: By checking the password in the "/:folder" endpoint, I think that means we would have to access the folder on the frontend and then submit yet again. Maybe something with app.get("/") would make more sense

File: backend/sample.js [hunk 3]
```diff
@@ -165,12 +193,11 @@ app.delete("/", (req, res) => {
...
-  let result = findNote(folderName, noteToDelete);
-  if (result === undefined || result.length == 0) {
-    res.status(404).send(noteToDelete);
-  } else {
-    deleteNote(folderName, noteToDelete);
+  const folderN = req.params["folderName"];
+  if (deleteNote(folderN, noteToDelete)) {
     res.status(204).end();
+  } else {
+    res.status(404).end();
```
Comment (matthewzelten) [inline backend/sample.js:200]: this delete section looks good to me

**Comment by matthewzelten** [comment]
Also this looks like theres gonna be conflicts

File: backend/sample.js (hunk 2 above)
Comment (Sahra-Zhou) [reply backend/sample.js:109]: Make sense. I'll improve this part.

**Pull request closed without merging**